### ISSUE TITLE
Remove the ldap configuration option to use the truststore spi

### ISSUE THREAD

**Issue by shawkins** [keycloak/keycloak#24663]
### Description

The UI and LDAPContextManager logic have an option to not use the spi truststore.  Once the work is completed on #24148 this should not be needed when using the --truststore-paths option.  Additional documetentation and/or deprecation should be considered. 

Even after #24148 is completed it will still be possible to override the system truststore with the spi-truststore-file-file option - if / when that is removed, then useTruststoreSpi can be removed as well.

cc @vmuzikar

**Comment by vmuzikar** [comment]
@keycloak/store Do you have any concerns about this?

**Comment by ahus1** [comment]
I agree this would simplify things for users to only use that CLI option. 

At the same time I don't really like the UX of the current and future behavior: I need to specify truststores on the CLI. When I as a user add LDAP via UI, I would expect to be able to upload a truststore for that specific connection I'm just now configuring. I've seen a similar UX for this for example when configuring tools like Sonatype Nexus or others.

Especially in an enterprise context where those truststores are used for selfsigned certificates, it makes IMHO sense to limit a trustustore to a single connection: I want to accept those self-signed certificates on a per connection (or per realm) basis, but not globally. 

@keycloak/core - as you've been involved in LDAP as well, please comment here as well.

**Comment by vmuzikar** [comment]
@ahus1 It's a good point, global server truststore is not enough e.g. for multitenancy scenarios where a single KC instance is used by multiple customers.

However truststore per connection/resource might be a bit too complex (Keycloak is doing a lots of various outgoing connections). But in the future we might consider a Realm truststore on top of the global system truststore (that'd serve as fallback for anything that doesn't use this, e.g. DB drivers etc.).

**Comment by rmartinc** [comment]
If I understand #24148 OK there are going to be multiple CA certs added to the trust-store on top of the JDK/default ones (if wanted). So yes, I think that the option `Use Truststore SPI` in LDAP is not necessary, as the new feature covers the two scenarios that this option manages right now (use default JDK or truststore SPI).

I also like the @ahus1 idea of managing individual trust-stores, but as @vmuzikar says it can be considered as a subsequent enhancement. The idea of having a trust-store per realm is very nice too.
